spec/requests/schools/partnerships_spec.rb

```diff
@@ -46,20 +56,19 @@
       end
 
       context "when the school has recently entered a partnership" do
-        before do
-          PartnershipNotificationEmail.create!(
-            token: "abc123",
-            sent_to: user.email,
-            partnership: @partnership,
-            email_type: PartnershipNotificationEmail.email_types[:induction_coordinator_email],
-          )
+        let!(:partnership) do
+          create(:partnership, :in_challenge_window,
+                 cohort: cohort,
+                 lead_provider: lead_provider,
+                 school: school,
+                 delivery_partner: delivery_partner1)
```

Comment:
I know you haven't done it, but is this context _inside_ `when the school is in a partnership` context? is our `let!(:partnership)` here overriding the one above? Can we split them up?

Reply:
I think this is one of the examples where I'd just love to split the view into components and check them being rendered instead of testing the whole html response with all the possible cases. But that's for later.

I have renamed this inner context for now - this is really a special case of the outer context (the partnership being still in the challenge window). As per splitting, for me it is an everlasting battle between well defined contexts and too-many-nested levels (and I am on both sides at the same time). I am still reusing quite a lot of the parent context, so I'd leave it as is (with the inner context renamed), but I'll need to read through those tests at some point as the setup seems a bit much (why do we need 5 delivery partners, what's the other partnership needed for here, why do we need ProviderRelationship on the school partnerships page?)